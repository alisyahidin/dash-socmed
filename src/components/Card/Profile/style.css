.profile-avatar {
  margin-bottom: 0.5rem;
}

.profile-username {
  color: #888;
}