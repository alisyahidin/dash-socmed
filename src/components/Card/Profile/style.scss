.profile-card {
  &__avatar {
    margin-bottom: 0.5rem;
  }
  &__username {
    color: #888;
  }
}