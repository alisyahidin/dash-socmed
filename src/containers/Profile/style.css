.profile-tabs {
  width: 100%;
  border-bottom: 1px solid #dee2e6 !important;
}

.profile-tabs > a {
  color: rgba(0, 0, 0, 0.5);
}

.tab-content {
  width: 100%;
}

.nav-pills > .nav-link.active {
  color: #FFF !important;
}