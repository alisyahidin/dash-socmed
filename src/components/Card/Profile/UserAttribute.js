import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-bootstrap'
import './style.scss'

const UserAttribute = ({label, value}) => {
  return (
    <div className="d-flex w-100">
      <Form.Label column sm="4">
        <b>{label}</b>
      </Form.Label>
      <Form.Label column>
        {value}
      </Form.Label>
    </div>
  )
}

UserAttribute.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
}

UserAttribute.defaultProps = {
  label: 'label',
  value: 'value',
}

export default UserAttribute