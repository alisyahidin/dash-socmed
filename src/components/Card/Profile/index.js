import React from 'react'
import PropTypes from 'prop-types'
import Avatar from 'react-avatar'

import Card from '../'
import UserAttribute from './UserAttribute'
import './style.scss'

const Profile = ({user}) => {
  const {address, company, id, name, username, posts, ...attributes} = user
  const {geo, ...listAddress} = address

  return (
    <Card align="left">
      <div className="mx-auto mb-2 d-flex flex-column align-items-center">
        <Avatar round size="70px" name={user.name} className="profile-card__avatar" />
        <h4>{name}</h4>
        <p className="profile-card__username">@{username}</p>
      </div>
      {Object.keys(attributes).map(label => (
        <UserAttribute key={label} label={label} value={attributes[label]} />
      ))}
      <b className="mt-2 mx-auto">Address</b>
      {Object.keys(listAddress).map(label => (
        <UserAttribute key={label} label={label} value={listAddress[label]} />
      ))}
      <b className="mt-2 mx-auto">Company</b>
      {Object.keys(company).map(label => (
        <UserAttribute key={label} label={label} value={company[label]} />
      ))}
    </Card>
  )
}

Profile.propTypes = {
  user: PropTypes.object.isRequired,
}

Profile.defaultProps = {
  user: {
    "id": 1,
    "name": "Guest",
    "username": "guest",
    "address": {
      "geo": {
        "lat": "-37.3159",
        "lng": "81.1496"
      }
    },
    "company": {
      "name": "Romaguera-Crona",
      "catchPhrase": "Multi-layered client-server neural-net",
      "bs": "harness real-time e-markets"
    }
  }
}

export default Profile