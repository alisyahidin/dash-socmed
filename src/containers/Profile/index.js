import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  Container,
  Row,
  Col,
  Tabs,
  Tab,
  Nav,
} from 'react-bootstrap'

import { fetchSingleUser, clearSingleUser } from '../../actions/singleUser'

import withNavbar from '../../hoc/withNavbar'
import ProfileCard from '../../components/Card/Profile/'
import PostCard from '../../components/Card/Post/'
import PostDetail from '../../components/PostDetail/'
import Card from '../../components/Card/'
import Image from '../../components/Image/'
import './style.scss'

import img from '../../assets/images/placeholder.png'

class Profile extends Component {
  componentDidMount() {
    const { location, fetchSingleUser, match } = this.props

    if (location.pathname !== '/profile') {
      fetchSingleUser(match.params.id)
    } else {
      fetchSingleUser(1)
    }
  }

  componentWillUnmount() {
    this.props.clearSingleUser()
  }

  render() {
    const { singleUser } = this.props

    return (
      <>
        <Container>
          <Row>
            <Col md={5}>
              {singleUser.data !== null
                ? <ProfileCard user={singleUser.data} />
                : <Card><h1>Loading</h1></Card>
              }
            </Col>
            <Col>
              <Card align="left" className="mb-sm-1">
                <Tabs defaultActiveKey="posts" className="profile-tabs m-sm-0">
                  <Tab eventKey="posts" title="Posts">
                    {singleUser.data === null && (
                      <h1 className="text-center">Loading</h1>
                    )}
                    {singleUser.data !== null && singleUser.data.posts.map((data, index) => (
                      <PostCard profileDisabled author={singleUser.data || {name: 'Guest', username: 'guest'}} key={index} {...data} />
                    ))}
                  </Tab>
                  <Tab eventKey="photos" title="Photos">
                    <Tab.Container id="albums" defaultActiveKey="first">
                      <Row className="m-sm-0 mt-sm-4">
                        <Col sm={3}>
                          <Nav variant="pills" className="flex-column">
                            <Nav.Item>
                              <Nav.Link eventKey="first">Album 1</Nav.Link>
                            </Nav.Item>
                            <Nav.Item>
                              <Nav.Link eventKey="second">Album 2</Nav.Link>
                            </Nav.Item>
                          </Nav>
                        </Col>
                        <Col sm={9}>
                          <Tab.Content>
                            <Tab.Pane eventKey="first">
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                            </Tab.Pane>
                            <Tab.Pane eventKey="second">
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                              <Image className="w-25" src={img} thumbnail />
                            </Tab.Pane>
                          </Tab.Content>
                        </Col>
                      </Row>
                    </Tab.Container>
                  </Tab>
                </Tabs>
              </Card>
            </Col>
          </Row>
        </Container>
        <PostDetail />
      </>
    )
  }
}

const mapStateToProps = state => state

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchSingleUser,
    clearSingleUser
  }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(withNavbar(Profile)))
