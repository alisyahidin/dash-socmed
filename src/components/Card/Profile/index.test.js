import React from 'react'
import { shallow, mount } from 'enzyme'
import Profile from './index'

it('render <Profile /> correctly', () => {
  const component = shallow(<Profile />)

  expect(component).toMatchSnapshot()
})

it('check <Profile /> props', () => {
  const component = mount(<Profile />)

  expect(component.props().user.name).toBe('Guest')
  expect(component.props().user.username).toBe('guest')
})
