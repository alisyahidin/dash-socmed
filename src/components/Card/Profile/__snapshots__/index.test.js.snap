// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Profile /> correctly 1`] = `
<Card
  align="left"
>
  <div
    className="mx-auto mb-2 d-flex flex-column align-items-center"
  >
    <withAvatarConfig
      className="profile-card__avatar"
      name="Guest"
      round={true}
      size="70px"
    />
    <h4>
      Guest
    </h4>
    <p
      className="profile-card__username"
    >
      @
      guest
    </p>
  </div>
  <b
    className="mt-2 mx-auto"
  >
    Address
  </b>
  <b
    className="mt-2 mx-auto"
  >
    Company
  </b>
  <UserAttribute
    key="name"
    label="name"
    value="Romaguera-Crona"
  />
  <UserAttribute
    key="catchPhrase"
    label="catchPhrase"
    value="Multi-layered client-server neural-net"
  />
  <UserAttribute
    key="bs"
    label="bs"
    value="harness real-time e-markets"
  />
</Card>
`;
